File: src/main/java/com/claudio_personagens/bookingcharactersapi/repositories/ClientRepository.java
package com.claudio_personagens.bookingcharactersapi.repositories;

import com.claudio_personagens.bookingcharactersapi.base.AbstractRepository;
import com.claudio_personagens.bookingcharactersapi.domain.Client;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

import java.util.Optional;

public interface ClientRepository extends AbstractRepository<Client, String>, JpaSpecificationExecutor<Client> {
    Optional<Client> findByName(String name);
}

File: src/main/java/com/claudio_personagens/bookingcharactersapi/repositories/ClientRepository.kt
package com.claudio_personagens.bookingcharactersapi.repositories

import com.claudio_personagens.bookingcharactersapi.base.BaseRepository
import com.claudio_personagens.bookingcharactersapi.domain.Client
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ClientRepository : BaseRepository<Client, UUID> {
    fun findByName(name: String): Optional<Client>
}